lusidtools/cocoon/cocoon.py

```diff
@@ -328,6 +330,19 @@ def _convert_batch_to_models(
     # Get the data types of the columns to be added as properties
     property_dtypes = data_frame.loc[:, property_columns].dtypes
 
+    if kwargs["sub_holding_key_type"] == "key-values":
```

Comment:
what is the `key-values` magic string? probably should be in the doc string

Reply:
It is a flag which specifies if the sub-holding-keys should be a dictionary of property values (holdings) or just a list of property keys (portfolio). 

The way that we tell originally is by doing

open_api_types = getattr(lusid.models, domain_lookup[file_type]["top_level_model"]).openapi_types

    if "sub_holding_keys" in open_api_types.keys() and "dict" in open_api_types["sub_holding_keys"]:

We could just do this again here instead of passing through this magic keyword argument, what do you think?

Reply:
I added some comments around the code for now. Haven't put anything in the DocString as I'm not sure on the best way to list kwargs in there.

Reply:
I think the original version might be better as it prevents the user from having to know about inner workings and ergo reduces scope for errors and makes it easier to use

Reply:
Okay done